src/main/java/io/managed/services/test/client/kafkainstance/KafkaInstanceApi.java

```diff
@@ -1,115 +1,151 @@
+import com.openshift.cloud.api.kas.auth.api.v1.V1RequestBuilder;
 import com.openshift.cloud.api.kas.auth.models.AclBinding;
 import com.openshift.cloud.api.kas.auth.models.AclBindingListPage;
-import com.openshift.cloud.api.kas.auth.models.AclBindingOrderKey;
-import com.openshift.cloud.api.kas.auth.models.AclOperationFilter;
-import com.openshift.cloud.api.kas.auth.models.AclPatternTypeFilter;
-import com.openshift.cloud.api.kas.auth.models.AclPermissionTypeFilter;
-import com.openshift.cloud.api.kas.auth.models.AclResourceTypeFilter;
+import com.openshift.cloud.api.kas.auth.models.AclOperation;
+import com.openshift.cloud.api.kas.auth.models.AclPatternType;
+import com.openshift.cloud.api.kas.auth.models.AclPermissionType;
+import com.openshift.cloud.api.kas.auth.models.AclResourceType;
 import com.openshift.cloud.api.kas.auth.models.ConsumerGroup;
-import com.openshift.cloud.api.kas.auth.models.ConsumerGroupDescriptionOrderKey;
 import com.openshift.cloud.api.kas.auth.models.ConsumerGroupList;
-import com.openshift.cloud.api.kas.auth.models.ConsumerGroupOrderKey;
 import com.openshift.cloud.api.kas.auth.models.NewTopicInput;
-import com.openshift.cloud.api.kas.auth.models.SortDirection;
 import com.openshift.cloud.api.kas.auth.models.Topic;
-import com.openshift.cloud.api.kas.auth.models.TopicOrderKey;
 import com.openshift.cloud.api.kas.auth.models.TopicSettings;
 import com.openshift.cloud.api.kas.auth.models.TopicsList;
 import io.managed.services.test.client.BaseApi;
 import io.managed.services.test.client.exception.ApiGenericException;
 import io.managed.services.test.client.exception.ApiUnknownException;
+import lombok.extern.log4j.Log4j2;
+import java.util.concurrent.TimeUnit;
 
+@Log4j2
 public class KafkaInstanceApi extends BaseApi {
 
     private final ApiClient apiClient;
-    private final AclsApi aclsApi;
-    private final GroupsApi groupsApi;
-    private final TopicsApi topicsApi;
+    private final V1RequestBuilder v1;
 
     public KafkaInstanceApi(ApiClient apiClient, String offlineToken) {
         super(offlineToken);
         this.apiClient = apiClient;
-        this.aclsApi = new AclsApi(apiClient);
-        this.groupsApi = new GroupsApi(apiClient);
-        this.topicsApi = new TopicsApi(apiClient);
+        this.v1 = apiClient.api().v1();
     }
 
     @Override
     protected ApiUnknownException toApiException(Exception e) {
-        if (e instanceof ApiException) {
-            var ex = (ApiException) e;
-            return new ApiUnknownException(ex.getMessage(), ex.getCode(), ex.getResponseHeaders(), ex.getResponseBody(), ex);
+        log.info(e);
+
+        if (e.getCause() != null) {
+            if (e.getCause() instanceof com.openshift.cloud.api.kas.auth.models.Error) {
+                var err = (com.openshift.cloud.api.kas.auth.models.Error) e.getCause();
+                return new ApiUnknownException(err.getReason(), err.getCode().toString(), err.responseStatusCode, err.getHref(), err.getId(), err);
```

Comment:
this doesn't look right, if we have a `models.Error` it means that it's one of the expected output of the endpoint and is not "Unknown"

Reply:
@henryZrncik I have made changes to return ApiGenericException for the `toApiException` abstract. I think this makes sense since the BaseAPI.handle method converted `ApiUnknownException` to `ApiGenericException`